### PR TITLE
Workflow - Build Docs

### LINKED ISSUE #1828: GitHub Actions - Workflow Build Docs
GitHub Actions workflow for build_docs.

Quip Doc: https://alteryx.quip.com/BAdaAzZljlRJ/Workflow-builddocs

### PR DESCRIPTION
**Pull request by ParthivNaresh**
Fixes #1828

### CODE REVIEW

**Review by freddyaboulton — APPROVED** [review]
@ParthivNaresh Looks great! I just have some non-blocking questions for my understanding.

**Review by bchen1116 — APPROVED** [review]
LGTM!

**Review by chukarsten — APPROVED** [review]
Looks good!

File: .github/workflows/build_docs.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+      - name: Set up Python ${{ matrix.python_version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python_version }}
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
```
[the remark this answers is not in the document]
Comment (ParthivNaresh) [reply .github/workflows/build_docs.yml:26]: They point to the same value in this case, but I believe if someone forks the repo and tries to submit a PR, `${{ github.event.pull_request.head.repo.full_name }}` will not point to the same value as `${{ github.repository }}`.

File: .github/workflows/build_docs.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: Build Docs
```
[the remark this answers is not in the document]
Comment (ParthivNaresh) [reply .github/workflows/build_docs.yml:1]: I'm testing a Slack integration action now!

File: .github/workflows/build_docs.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (freddyaboulton) [reply .github/workflows/build_docs.yml:26]: Cool, thanks for explaining!

**Comment by dsherry** [comment]
@ParthivNaresh given that we decided not delete the `build_docs` job rather than to merge this PR, could you please close this PR?

**Pull request closed without merging**